atomics/include/desul/atomics/Compare_Exchange_SYCL.hpp

```diff
@@ -20,86 +20,76 @@ SPDX-License-Identifier: (BSD-3-Clause)
+template <class T, class MemoryOrder, class MemoryScope>
+std::enable_if_t<sizeof(T) == 4, T> host_atomic_exchange(T* const dest,
+                                                         T value,
+                                                         MemoryOrder,
+                                                         MemoryScope) {
   static_assert(sizeof(unsigned int) == 4,
                 "this function assumes an unsigned int is 32-bit");
-  Impl::sycl_atomic_ref<unsigned int, MemoryOrder, MemoryScope> dest_ref(
+  sycl_atomic_ref<unsigned int, MemoryOrder, MemoryScope> dest_ref(
       *reinterpret_cast<unsigned int*>(dest));
   unsigned int return_val = dest_ref.exchange(*reinterpret_cast<unsigned int*>(&value));
   return reinterpret_cast<T&>(return_val);
 }
-template <typename T, class MemoryOrder, class MemoryScope>
-typename std::enable_if<sizeof(T) == 8, T>::type atomic_exchange(T* const dest,
-                                                                 T value,
-                                                                 MemoryOrder,
-                                                                 MemoryScope) {
+
+template <class T, class MemoryOrder, class MemoryScope>
+std::enable_if_t<sizeof(T) == 8, T> host_atomic_exchange(T* const dest,
+                                                         T value,
+                                                         MemoryOrder,
+                                                         MemoryScope) {
   static_assert(sizeof(unsigned long long int) == 8,
                 "this function assumes an unsigned long long is 64-bit");
-  Impl::sycl_atomic_ref<unsigned long long int, MemoryOrder, MemoryScope> dest_ref(
+  sycl_atomic_ref<unsigned long long int, MemoryOrder, MemoryScope> dest_ref(
       *reinterpret_cast<unsigned long long int*>(dest));
   unsigned long long int return_val =
       dest_ref.exchange(reinterpret_cast<unsigned long long int&>(value));
   return reinterpret_cast<T&>(return_val);
 }
 
-template <typename T, class MemoryOrder, class MemoryScope>
-typename std::enable_if<(sizeof(T) != 8) && (sizeof(T) != 4), T>::type
-atomic_compare_exchange(
+template <class T, class MemoryOrder, class MemoryScope>
+std::enable_if_t<(sizeof(T) != 8) && (sizeof(T) != 4), T> host_atomic_compare_exchange(
     T* const /*dest*/, T compare, T /*value*/, MemoryOrder, MemoryScope) {
-  // FIXME_SYCL not implemented
-  assert(false);
+  assert(false);  // FIXME_SYCL not implemented
   return compare;
 }
 
-template <typename T, class MemoryOrder, class MemoryScope>
-typename std::enable_if<(sizeof(T) != 8) && (sizeof(T) != 4), T>::type atomic_exchange(
+template <class T, class MemoryOrder, class MemoryScope>
+std::enable_if_t<(sizeof(T) != 8) && (sizeof(T) != 4), T> host_atomic_exchange(
     T* const /*dest*/, T value, MemoryOrder, MemoryScope) {
-  // FIXME_SYCL not implemented
-  assert(false);
```

Comment:
Any reason to change this? It doesn't seem like trailing `FIXME` comments are consistently applied anyway.

Reply:
Are you referring to moving the comment to the end of the `assert(false)` line?
(Assuming that was the question) because I find it more compact and grep-friendly.
What inconsistency are you referring to.  This is consistent at the file scope if anything.

Reply:
Yes, I was referring to that. 
It might be that [atomics/include/desul/atomics/Compare_Exchange_OpenMP.hpp](https://github.com/desul/desul/pull/69/files#diff-e9d0a57870cabfd9e6d97723c946a7bd7f3f4c82cf1a39045725db575a32d676)
```
  // FIXME_OPENMP make sure this never gets called
  return value;
```
 is the only file where you could also move the comment to the end of the line referred to. My preference still is to have it above the respective line for readability but I can also see your point. Nothing worth holding this pull request up over.